src/webgpu/api/validation/image_copy/buffer_related.spec.ts

```diff
@@ -104,8 +104,15 @@ g.test('bytes_per_row_alignment')
       .beginSubcases()
       .combine('bytesPerRow', [undefined, 0, 1, 255, 256, 257, 512])
       .combine('copyHeight', [0, 1, 2, 4, 8])
-      // copyHeight must be aligned with blockHeight
-      .filter(({ format, copyHeight }) => copyHeight % kTextureFormatInfo[format].blockHeight === 0)
+      // copyHeight has some constraints:
+      //   - it must be aligned with blockHeight.
+      //   - if it is 0, the format cannot be a depth/stencil format. Because it must be a full copy if the format is a depth/stencil format but copied subresource's height can never be 0.
```

Comment:
This restriction comes in because of how copyHeight=0 is specially treated (that it's rounded up to blockHeight). It's confusing here without reading the test even though it's explained by the comment.

That's ok, but maybe possible to do something like:

```ts
.combine('copyHeightInBlocks', [0, 1, 2, 4, 8])
.expand('_textureHeightInBlocks', p => p.copyHeightInBlocks === 0 ? 1 : p.copyHeightInBlocks)
// Depth/stencil format copies must copy the whole subresource.
.unless(p => {
  const info = kTextureFormatInfo[p.format];
  return (info.depth || info.stencil) && p.copyHeightInBlocks !== p._textureHeightInBlocks;
})
```

Reply:
Thanks for the suggestion, Kai. I tried the code, looks like `expand` requires the expanded parameter `_textureHeightInBlocks` to be iterable but it is not in the code snippet. So it failed to build. 

Any suggestion? Maybe we can just merge this PR?

Reply:
Sorry, that was missing a pair of brackets.

`.expand('_textureHeightInBlocks', p => [p.copyHeightInBlocks === 0 ? 1 : p.copyHeightInBlocks])`

Reply:
Done. I should have figured out this.